RELEASE.md

```diff
@@ -1,31 +1,72 @@
 # How to release Kuadrant Operator
 
-## Process
+## Kuadrant Operator Release
```

Comment:
Probably merge this PR after https://github.com/Kuadrant/kuadrant-operator/pull/1150

Reply:
Sorry worry branch target for the merge. Trying the `gh` cli lately to create PRs and never though about which branch it was being opened against.